unitex/src/fr/umlv/unitex/utils/SingleInstanceMonitor.java

```diff
@@ -0,0 +1,248 @@
+package fr.umlv.unitex.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.net.BindException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.util.ArrayList;
+
+import javax.swing.JFrame;
+import javax.swing.SwingWorker;
+
+import fr.umlv.unitex.Unitex;
+import fr.umlv.unitex.config.Config;
+import fr.umlv.unitex.frames.UnitexFrame;
+
+
+/**
+ * This class ensures only one instance of the IDE can run at one time
+ * @author markpower
+ *
+ */
+public class SingleInstanceMonitor {
+
+	private ServerSocket serverSocket = null;
+	private Socket socket = null;
+	private final String DELIMITER = ";";
+	private boolean SOCKET_OPEN = false;
+	private final String PORT_FILE_NAME = "single_instance.port";
+	
+	
+	public boolean isRunning(final String[] args) {
+    	try {
+    		// set the app path so we know where to store the port file
+    		Config.setAppPath(args);
+    		
+    		File f = getPortFile();
+			
+			if (f.exists()) {
```

Comment:
Under Linux with Java 8 (1.8.0_121) it's possible two run multiple instances following the next steps:

First instance:
```
mkdir /tmp/test
mv Unitex.jar Gramlab.jar /tmp/test
cd /tmp/test
java -jar Unitex.jar
```

Second instance:
```
cd /tmp/test
java -jar Unitex.jar -Dunitex.binary.dir="/path/to/real/Unitex-GramLab-3.1/App" foo.grf
```